src/handler/core.rs

```diff
@@ -122,6 +236,25 @@ where
     /// chaining and
     /// branching"](#the-difference-between-chaining-and-branching).
     ///
+    /// # Type inference algorithm
```

Comment:
You should replace all mentions of "inference" with "checking", we don't actually do any inference.

Reply:
We do dynamic type inference inside `chain` and `branch` and checking inside `type_check`, but it's not unification-based type inference, of course.